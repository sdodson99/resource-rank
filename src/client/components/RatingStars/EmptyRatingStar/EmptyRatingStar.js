import React from 'react';
import PropTypes from 'prop-types';
import styles from './EmptyRatingStar.module.css';
import { StarRounded } from '@material-ui/icons';

const EmptyRatingStar = ({ size }) => (
  <div className={styles.EmptyRatingStar} data-testid="EmptyRatingStar">
    <StarRounded htmlColor={'#d6d6d6'} style={{ fontSize: size }} />
  </div>
);

EmptyRatingStar.propTypes = {
  size: PropTypes.number,
};

EmptyRatingStar.defaultProps = {
  size: 20,
};

export default EmptyRatingStar;
