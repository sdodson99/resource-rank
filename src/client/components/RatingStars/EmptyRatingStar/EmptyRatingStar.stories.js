import React from 'react';
import EmptyRatingStar from './EmptyRatingStar';

export default {
  component: EmptyRatingStar,
  title: 'Components/RatingStars/EmptyRatingStar',
};

const Template = (args) => <EmptyRatingStar {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Size = Template.bind({});
Size.args = {
  size: '100',
};
