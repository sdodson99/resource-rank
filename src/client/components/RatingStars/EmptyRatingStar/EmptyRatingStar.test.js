import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import EmptyRatingStar from './EmptyRatingStar';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<EmptyRatingStar />', () => {
  it('should mount', () => {
    render(<EmptyRatingStar />);

    const emptyRatingStar = screen.getByTestId('EmptyRatingStar');

    expect(emptyRatingStar).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<EmptyRatingStar />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with custom size', () => {
    const component = createRenderer().render(<EmptyRatingStar size={100} />);

    expect(component).toMatchSnapshot();
  });
});
