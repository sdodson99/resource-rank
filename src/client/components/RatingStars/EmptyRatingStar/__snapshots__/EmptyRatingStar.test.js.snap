// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyRatingStar /> should render correctly 1`] = `
<div
  className="EmptyRatingStar"
  data-testid="EmptyRatingStar"
>
  <Memo
    htmlColor="#d6d6d6"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</div>
`;

exports[`<EmptyRatingStar /> should render correctly with custom size 1`] = `
<div
  className="EmptyRatingStar"
  data-testid="EmptyRatingStar"
>
  <Memo
    htmlColor="#d6d6d6"
    style={
      Object {
        "fontSize": 100,
      }
    }
  />
</div>
`;
